src/api/status/public/js/build-log/datetime-calculator.js

```diff
@@ -0,0 +1,45 @@
+const buildDateCaltulator = document.getElementById('calculatorData');
+const buildLoading = document.getElementById('build-loading');
+const buildSuccess = document.getElementById('build-success');
+const buildFail = document.getElementById('build-fail');
+
+export default function calculatorBuildInfo(data) {
+  const startedAtValue = data.startedAt;
+  const stoppedAtValue = data.stoppedAt;
+  const resultValue = data.result;
+
+  if (startedAtValue !== 'null' && stoppedAtValue === 'null') {
+    const date = Math.abs((new Date(startedAtValue).getTime() / 1000).toFixed(0));
+    const currentDate = Math.abs((new Date().getTime() / 1000).toFixed(0));
+    const diff = currentDate - date;
+    const hours = Math.floor(diff / 3600) % 24;
+    const minutes = Math.floor(diff / 60) % 60;
+    const seconds = diff % 60;
```

Comment:
This should be extracted into a function. It's used three times here for the same calculation.